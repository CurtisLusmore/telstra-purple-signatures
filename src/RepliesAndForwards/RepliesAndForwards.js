import React from 'react';

import { purple } from '../constants';
import { parseMobile, parseLandLine } from '../util';
const { brandInfo } = purple;

const RepliesAndForwards = props => {
  const {
    name,
    title,
    mobile,
    email,
    isSupport,
    twitter,
    supportHotline,
    supportEmail
  } = props;
  const styleObj = {
    color: 'black',
    fontFamily: 'Arial, sans-serif',
    fontSize: '9pt'
  };
  return (
    <div>
      <p style={styleObj}>
        --
        <br />
        <b>
          {name}&nbsp;|&nbsp;{brandInfo.brandName}&nbsp;|&nbsp;{title}
        </b>
        <br />
        <b>M</b>&nbsp;
        <a
          href={`tel:${parseMobile(mobile).replace(/&nbsp;/g, '')}`}
          dangerouslySetInnerHTML={{ __html: parseMobile(mobile) }}
          style={{ color: brandInfo.brandLinkColour }}
        />{' '}
        | <b>E</b>&nbsp;
        <a
          href={`mailto:${email}`}
          style={{ color: brandInfo.brandLinkColour }}
        >
          {email}
        </a>{' '}
        |{' '}
        {twitter ? (
          <span>
            <span>
              <b>T</b>&nbsp;
              <a
                href={`https://twitter.com/${twitter.replace('@', '')}`}
                rel="noopener noreferrer"
                target="_blank"
                style={{ color: brandInfo.brandLinkColour }}
              >
                @{twitter.replace('@', '')}
              </a>
            </span>{' '}
            |{' '}
          </span>
        ) : null}
        <b>W</b>&nbsp;
        <a
          href={brandInfo.brandLink}
          style={{ color: brandInfo.brandLinkColour }}
        >
          {brandInfo.brandLinkName}
        </a>
        {isSupport ? (
          <span>
            <br />
            <b>Support&nbsp;Hotline</b>&nbsp;
            <a
              href={`tel:${parseLandLine(supportHotline).replace(
                /&nbsp;/g,
                ''
              )}`}
              dangerouslySetInnerHTML={{
                __html: parseLandLine(supportHotline)
              }}
              style={{ color: brandInfo.brandLinkColour }}
            />{' '}
            | <b>Support&nbsp;Email</b>&nbsp;
            <a
              href={`mailto:${supportEmail}`}
              style={{ color: brandInfo.brandLinkColour }}
            >
              {supportEmail}
            </a>
          </span>
        ) : null}
      </p>
      <div
        style={{
          fontSize: '6.5pt',
          lineHeight: '1.5'
        }}
      >
        This email may contain confidential information.
        <br />
        If I've sent it to you by accident, please delete it immediately.
      </div>
    </div>
  );
};

export default RepliesAndForwards;
