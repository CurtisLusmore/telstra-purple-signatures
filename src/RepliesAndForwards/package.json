{
  "main": "./RepliesAndForwards.js",
  "name": "RepliesAndForwards",
  "version": "0.0.0",
  "private": true
}