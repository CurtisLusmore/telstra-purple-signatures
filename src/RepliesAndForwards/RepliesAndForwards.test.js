import React from 'react';
import { shallow } from 'enzyme';

import RepliesAndForwards from './RepliesAndForwards';

it('Renders', () => {
  const props = {
    emailHtml: <div>your.name@readify.net</div>,
    isSupport: false,
    mobileHtml: <div>+61 111 222 333</div>,
    name: 'your name',
    placeholders: {
      email: 'your.name@readify.net',
      mobile: '+61 111 222 333',
      name: 'Your Name',
      qualifications: 'Jedi Master | PSM I',
      title: 'Job Title',
      twitter: '@myTwitter',
    },
    title: 'Job Title',
    twitterHtml: <div>@myTwitter</div>,
    brandLink: 'https://test.com',
    brandName: 'Test',
    brandLinkName: 'Test Link Name'
  };

  const tree = shallow(<RepliesAndForwards {...props} />);
  expect(tree).toMatchSnapshot();
});