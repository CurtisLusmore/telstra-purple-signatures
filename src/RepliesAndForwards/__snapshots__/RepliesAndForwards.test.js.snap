// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders 1`] = `
<div>
  <p
    style={
      Object {
        "color": "#3D5567",
        "fontFamily": "Segoe UI, Tahoma, Geneva, Verdana, sans-serif",
        "fontSize": "10.5pt",
      }
    }
  >
    --
    <br />
    <b>
      your name
       | 
      Readify
       
      Job Title
    </b>
    <br />
    <b>
      M
    </b>
      
    <a
      dangerouslySetInnerHTML={
        Object {
          "__html": "+61&nbsp;111&nbsp;222&nbsp;333",
        }
      }
      href="tel:+61111222333"
    />
     | 
    <b>
      E
    </b>
      
    <a
      href="mailto:your.name@readify.net"
    >
      your.name@readify.net
    </a>
     | 
    <span>
      <span>
        @myTwitter
      </span>
      | 
    </span>
    <b>
      W
    </b>
     
    <a
      href="https://readify.net"
    >
      readify.net
    </a>
  </p>
</div>
`;
